classes/PaymentPreprocessor.js

```diff
@@ -0,0 +1,29 @@
+const {api} = require('actionhero')
+const isBefore = require('date-fns/is_before')
+
+module.exports = async function (payment) {
+  const processed = Object.assign({}, payment)
+  if (!processed.members) {
+    processed.members = []
+  }
+  const billingMember = await
+    api.models.member.findById(payment.billingMemberId)
+  if (!processed.members.includes(billingMember.id)) {
```

Comment:
why do you search the member in db, when you already have it's id and you use only that?